.github/workflows/main_ci.yml

```diff
@@ -0,0 +1,76 @@
+name: Python Testing
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - '*'
```

Comment:
we can narrow this down to `main` I think--broader now just for the demo tests on my fork against this branch/CI config